server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -478,7 +478,14 @@ private ClusterState applyCreateIndexWithTemporaryService(
             );
 
             indexService.getIndexEventListener().beforeIndexAddedToCluster(indexMetadata.getIndex(), indexMetadata.getSettings());
-            return clusterStateCreateIndex(currentState, request.blocks(), indexMetadata, allocationService::reroute, metadataTransformer);
+            BiFunction<ClusterState, String, ClusterState> rerouteFunction = (current, reason) -> {
```

Comment:
Just realized this: we never mutate the request later. So we could just do:

```java
rerouteFunction = request.performReroute() ? allocationService::reroute : (cs, reason) -> cs;
```

right?

Reply:
done: 51077a961b1c63a5bc955a676620e380ffb3e777